contributing.rst

```diff
@@ -0,0 +1,145 @@
+Contributing to nose2
+=====================
+
+Please do! nose2 cannot move forward without contributions from the
+testing community.
+
+This document is a set of guidelines, not strict rules.
+Use your best judgement, and feel free to propose changes to this document
+in a pull request.
+
+The Basics
+----------
+
+nose2 is hosted on `github`_. Our home there is
+https://github.com/nose-devs/nose2. We use github's issue tracking and
+collaboration tools *exclusively* for managing nose2's
+development. This means:
+
+* Please report issues here: https://github.com/nose-devs/nose2/issues
+
+* Please make feature requests in the same place
+
+* Please submit all patches as github pull requests
+
+Coding Guidelines
+-----------------
+
+The main rule is: *any patch that touches code must include tests.*
+And of course all tests must pass under all supported versions of Python.
+
+Fortunately that's easy to check: nose2 uses `tox`_ to manage its test
+scenarios, so simply running ``tox`` in nose2's root directory will
+run all of the tests with all supported python versions. When your
+patch gets all green, send a pull request!
+
+Some additional tips for the python and documentation in this project.
+
+- Code should be `pep8`_ compliant
+- Where possible, write code which passes ``pyflakes`` linting (consider using
+  ``flake8`` to do ``pyflakes`` and ``pep8`` checking)
+- For consistency with ``unittest`` please use CamelCase for class names,
+  methods, attributes and function parameters that map directly to class
+  attributes.
+- Try to use raw strings for docstrings -- ensures that ReST won't be
+  confused by characters like ``\\``
+- For complex functionality, include sample usage in docstrings
+- Comment liberally, but don't comment on every line of code
+- Use examples very liberally in documentation
+- Use double-quotes for strings, except when quoting a string containing
+  double-quotes but not containing single quotes
+- Use absolute imports everywhere
+- Avoid circular imports whenever possible -- given the choice between adding
+  a new module or adding a circular import, add the new module
+- Import non-``nose2`` modules and packages before importing from within
+  ``nose2``
+- Think very hard before adding a new dependency -- keep the dependencies of
+  ``nose2`` as lightweight as possible
+
+Commit Messages
+~~~~~~~~~~~~~~~
+
+A few basic ground rules for what ideal commits should look like.
+
+- No lines over 72 characters
+- No GitHub emoji -- use your words
+- Reference issues and pull requests where appropriate
+- Present tense and imperative mood
+
+Workflow and Merging
+--------------------
+
+Get Started
+~~~~~~~~~~~
+
+The ``bootstrap.sh`` script in the root of the nose2 distribution can be
```

Comment:
I assume you tested this...I don't use virtualenvwrapper. I install it and test next time I devote a block of time to this.